src/mavsdk/plugins/log_streaming/log_streaming_impl.cpp

```diff
@@ -254,46 +271,62 @@ void LogStreamingImpl::process_logging_data_acked(const mavlink_message_t& messa
     }
 
     std::lock_guard<std::mutex> lock(_mutex);
-    check_sequence(logging_data_acked.sequence);
-
-    if (logging_data_acked.first_message_offset == std::numeric_limits<uint8_t>::max()) {
-        _ulog_data.insert(
-            _ulog_data.end(),
-            logging_data_acked.data,
-            logging_data_acked.data + logging_data_acked.length);
-    } else {
-        if (logging_data_acked.first_message_offset > sizeof(logging_data_acked.data)) {
-            LogWarn() << "Invalid first_message_offset";
-            return;
-        }
-
-        _ulog_data.insert(
-            _ulog_data.end(),
-            logging_data_acked.data,
-            logging_data_acked.data + logging_data_acked.first_message_offset);
-        process_message();
-        _ulog_data.clear();
-
-        _ulog_data.insert(
-            _ulog_data.end(),
-            logging_data_acked.data + logging_data_acked.first_message_offset,
-            logging_data_acked.data + logging_data_acked.length);
+    auto drop_state = check_sequence(logging_data_acked.sequence);
+
+    switch (drop_state) {
+        case DropState::Ok:
+            if (logging_data_acked.first_message_offset == std::numeric_limits<uint8_t>::max()) {
+                _ulog_data.insert(
+                    _ulog_data.end(),
+                    logging_data_acked.data,
+                    logging_data_acked.data + logging_data_acked.length);
+            } else {
+                if (logging_data_acked.first_message_offset > sizeof(logging_data_acked.data)) {
+                    LogWarn() << "Invalid first_message_offset";
+                    return;
+                }
+
+                _ulog_data.insert(
+                    _ulog_data.end(),
+                    logging_data_acked.data,
+                    logging_data_acked.data + logging_data_acked.first_message_offset);
+                process_message();
+                _ulog_data.clear();
+
+                _ulog_data.insert(
+                    _ulog_data.end(),
+                    logging_data_acked.data + logging_data_acked.first_message_offset,
+                    logging_data_acked.data + logging_data_acked.length);
+            }
+            break;
+
+        case DropState::Dropped:
+            _ulog_data.clear();
+            _ulog_data.insert(
+                _ulog_data.end(),
+                logging_data_acked.data + logging_data_acked.first_message_offset,
+                logging_data_acked.data + logging_data_acked.length);
```

Comment:
Here you must only insert if there's a valid offset: `logging_data_acked.first_message_offset != std::numeric_limits<uint8_t>::max()`

Reply:
Oh, right!